core/src/ser.rs

```diff
@@ -860,8 +860,14 @@ pub trait AsFixedBytes: Sized + AsRef<[u8]> {
 	/// The length in bytes
 	fn len(&self) -> usize;
 }
-
-impl<'a> AsFixedBytes for &'a [u8] {
+// impl<'a> AsFixedBytes for &'a [u8] {
+// 	// TODO - This does not feel right?
+// 	fn len(&self) -> usize {
+// 		1
+// 	}
+// }
+impl AsFixedBytes for &[u8] {
+	// TODO - This does not feel right?
```

Comment:
Then what would be right?
Why do you need to implement a length function for byte arrays?

Reply:
Yeah this was just something I noticed when looking at this. Its not related to this PR so I'm going to revert these changes and discuss elsewhere - but something feels wrong with this code to me.

Reply:
Moved this to a separate PR. See #3131 for wider cleanup of `AsFixedBytes`.